src/zenml/client.py

```diff
@@ -2229,6 +2247,213 @@ def delete_stack_component(
             component.name,
         )
 
+    def deploy_stack_component(
+        self,
+        ctx: click.Context,
+        name: str,
+        flavor: str,
+        cloud: str,
+        component_type: StackComponentType,
+        configuration: Optional[Dict[str, Any]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+    ) -> "ComponentResponseModel":
+        """Deploys a stack component.
+
+        Args:
+            ctx: The click context.
+            name: The name of the deployed stack component.
+            flavor: The flavor of the deployed stack component.
+            cloud: The cloud of the deployed stack component.
+            component_type: The type of the stack component to deploy.
+            configuration: The configuration of the deployed stack component.
+            metadata: The metadata of the deployed stack component.
+
+        Returns:
+            The deployed stack component.
+        """
+        from zenml.cli.stack_recipes import (
+            deploy,
+            get_outputs,
+        )
+
+        STACK_COMPONENT_RECIPE_DIR = "deployed_stack_components"
+
+        if configuration:
+            # convert configuration dict to a json object
+            configuration = json.dumps(configuration)
+
+        # set the stack component and flavor
+        component = {component_type.value: flavor}
+
+        # path should be fixed at a constant in the
+        # global config directory
+        path = Path(
+            os.path.join(
+                io_utils.get_global_config_directory(),
+                STACK_COMPONENT_RECIPE_DIR,
+            )
+        )
+        # Create the new component model
+        result = ctx.invoke(
+            deploy,
+            path=path,
+            stack_recipe_name=f"{cloud}-modular",
+            config=configuration,
+            no_server=True,
+            **component,
+        )
```

Comment:
Not sure this feels right to be invoking a CLI command within the `client` code.

Reply:
Thinking about this more, we should probably rather be invoking whatever commands / class etc that the CLI is invoking here within its implementation rather than invoking the click command itself.

Reply:
I agree that it doesn't feel right but there's a lot of code that needs to be executed when you are about to deploy the recipe. Stuff like checking prerequisites, checking if existing service exists, not to mention error handling. In other words, everything in that deploy CLI function would have to be rewriten inside the client function which I think is not very good. Still open to changes however

Reply:
I think the idea is that you'd have some util function(s) / or class which deals with deployment that you can call from both places. I think you'd want to have all that functionality in one place that can be called independently of the CLI. Where that code lives I guess is an open question. (Maybe @schustmi you have thoughts on whether it should be on the client or separately as a different chunk of functionality.) But I think it's an opportunity to a) not have CLI code in the client as that's not a good pattern and b) have an abstract way to do everything that needs handling (everything you mention above).